File: app/src/main/java/com/nodag/app/ui/PortAdapter.java
package com.nodag.app.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.nodag.app.R;
import com.nodag.app.nodes.PortDirection;
import java.util.ArrayList;
import java.util.List;

/**
 * Адаптер для списка портов (входов/выходов)
 */
public class PortAdapter extends RecyclerView.Adapter<PortAdapter.ViewHolder> {

    private final PortDirection direction;
    private final OnDeleteClickListener onDeleteClick;
    private final List<PortItem> ports;

    public interface OnDeleteClickListener {
        void onDeleteClick(String portName);
    }

    public PortAdapter(PortDirection direction, OnDeleteClickListener onDeleteClick) {
        this.direction = direction;
        this.onDeleteClick = onDeleteClick;
        this.ports = new ArrayList<>();
    }

    public void addPort(PortItem port) {
        ports.add(port);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_port, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(ports.get(position));
    }

    @Override
    public int getItemCount() {
        return ports.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        private TextView portName;
        private TextView portType;
        private ImageButton btnDelete;

        ViewHolder(View itemView) {
            super(itemView);
            portName = itemView.findViewById(R.id.portName);
            portType = itemView.findViewById(R.id.portType);
            btnDelete = itemView.findViewById(R.id.btnDelete);
        }

        void bind(PortItem port) {
            portName.setText(port.getName());
            portType.setText(port.getType().name());
            btnDelete.setOnClickListener(v -> onDeleteClick.onDeleteClick(port.getName()));
        }
    }
}

File: app/src/main/java/com/nodag/app/ui/PortAdapter.kt
package com.nodag.app.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.nodag.app.R
import com.nodag.app.nodes.PortDirection

/**
 * Адаптер для списка портов (входов/выходов)
 */
class PortAdapter(
    private val direction: PortDirection,
    private val onDeleteClick: (String) -> Unit
) : RecyclerView.Adapter<PortAdapter.ViewHolder>() {

    private val ports = mutableListOf<PortItem>()

    fun addPort(port: PortItem) {
        ports.add(port)
        notifyDataSetChanged()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val portName: TextView = itemView.findViewById(R.id.portName)
        private val portType: TextView = itemView.findViewById(R.id.portType)
        private val btnDelete: ImageButton = itemView.findViewById(R.id.btnDelete)

        fun bind(port: PortItem) {
            portName.text = port.name
            portType.text = port.type.name

            btnDelete.setOnClickListener {
                onDeleteClick(port.name)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_port, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(ports[position])
    }

    override fun getItemCount(): Int = ports.size
}
